charmcraft/reactive_plugin.py

```diff
@@ -113,19 +115,24 @@ def get_build_environment(self) -> Dict[str, str]:
 
     def get_build_commands(self) -> List[str]:
         """Return a list of commands to run during the build step."""
+        options = cast(ReactivePluginProperties, self._options)
+
         command = [
             sys.executable,
             "-I",
             __file__,
             self._part_info.project_name,
             str(self._part_info.part_build_dir),
             str(self._part_info.part_install_dir),
+            json.dumps(options.reactive_charm_build_arguments),
```

Comment:
We can't just use whatever the user specified in the YAML, otherwise we would be sending "together" options that should be different parts of the command. 

IOW, if the user specified in the YAML...
```
    reactive-charm-build-arguments:
      - --layer-index 3
```
, we should NOT execute charm with `["charm", "build", "--layer-index 3", "-o", builddir]`, but with `["charm", "build", "--layer-index", "3", "-o", builddir]`.

You can use `shlex.split` to separate the strings from the YAML to be suitable for the command line usage.